code/modules/projectiles/ammunition/ammo_pile.dm

```diff
@@ -0,0 +1,166 @@
+/obj/item/ammo_pile
+	name = "ammo pile"
+	desc = "A handful of some kind of ammunition."
+	w_class = ITEMSIZE_SMALL
+
+	var/list/ammo = list()
+	var/list/ammo_overlays = list()
+	var/image/ammo_overlay = list()
+	var/ammo_type // the type of ammo this ammo pile accepts
+	var/max_ammo = 5
+
+/obj/item/ammo_pile/Initialize(mapload, ...)
+	. = ..()
+	if(ammo_type)
+		var/obj/first_round = new ammo_type(get_turf(src))
+		add_ammo(first_round)
+		check_name_and_ammo()
+		for(var/i = 1, i <= max_ammo - 1, i++)
```

Comment:
shouldn't this use the projectile's max_stack?

Reply:
it does, check_name_and_ammo sets max_ammo and properly names the pile